oops/oops.go

```diff
@@ -163,17 +179,24 @@ func isPrefix(a []uintptr, b []uintptr) bool {
 }
 
 func wrapf(err error, reason string) error {
-	var cause error
+	inner := err
 	var previous *oopsError
 
 	var st *stack
 	var index int
 	found := false
 
-	if e, ok := err.(*oopsError); ok {
-		cause = e.cause
+	// Find the previous error in our input, if any.
+	var e *oopsError
+	if ok := As(err, &e); ok {
 		previous = e
 
+		// If the input error was an oops error, then we set the cause for ourselves
+		// to the same error. Otherwise, we keep the input error as our own cause.
+		if _, ok := err.(*oopsError); ok {
```

Comment:
Don't we already know it's an oopsError at this point?

Reply:
Hmm, the comment is saying that the input error itself is an oops error (e.g. from `wrapf(err, reason)`). The error value that we know is oops is the `e` value. This is special casing the flow where we are appending to the oops chain instead of wrapping something intermediary (see the diagram in the PR desc). If we didn't guard the check, that would mean the outer most oops would point to the inner most base error.

Is there something to reword to make this more clear?

Reply:
Ahh gotcha. "If the input error itself is not an oopsError, we should keep inner as the input error even if it wraps other oopsErrors" is how I grok this.

Reply:
i updated the comment here 👍